### PR TITLE
Update podspecs to min iOS 10

### LINKED ISSUE #4847: Update minimum supported iOS version to 10
We're planning to update the minimum supported ios version to 10 in the next Firebase major release. 

Also do a scrub to remove unnecessary pre-iOS 10 code like removing NSNotificationCenter observers in dealloc's. See cl/307613047.

```
- (void)dealloc {
  [[NSNotificationCenter defaultCenter] removeObserver:self];
}
```

Update the AdMob dependency in the Firebase pod past 7.63. See #6355

Please comment here if you have any question or concerns.

### PR DESCRIPTION
**Pull request by paulb777**
- Update minimum iOS version across project to iOS 10
- Only exceptions are Analytics dependencies which are update to iOS 9 to be consistent with AdMob
- Only library source change is a code deletion from FDL to fix a deprecation warning
- Similar test change in FIAM
- Also osx 10.12 to be consistent with minimum Apple versions at 2016

Fix #4847

### CODE REVIEW

File: FirebaseCore/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# Firebase 7.0.0
+- [changed] Update minimum iOS version to iOS 10 except for Analytics which is now iOS 9. (#4847)
+- [changed] Update minimum OSX version to 10.12.
```
Comment (wilhuff) [inline FirebaseCore/CHANGELOG.md:3]: Apple no longer calls this operating system "OSX": it's "macOS".

**Review by wilhuff — APPROVED** [review]
LGTM

**Review by maksymmalyhin — APPROVED** [review]
LGTM

**Comment by paulb777** [comment]
All tests passed on the penultimate run and are failing after a changelog update because of CocoaPods CDN flakes.  

I'm going to merge now and can address additional comments on a subsequent PR.

**Pull request merged**